engine/src/main/java/org/terasology/logic/players/PlayerSystem.java

```diff
@@ -88,6 +97,15 @@ public void update(float delta) {
                 i.remove();
             }
         }
+
+        i = clientsPreparingToRespawn.iterator();
+        while (i.hasNext()) {
+            SpawningClientInfo spawning = i.next();
+            if (worldProvider.isBlockRelevant(spawning.position)) {
+                respawnPlayer(spawning.clientEntity);
+                i.remove();
```

Comment:
Is this vulnerable to `ConcurrentModificationException` due to the iterator being modified while iterating?